tests/cpp/test_greedy.cpp

```diff
@@ -288,4 +288,33 @@ TEST_F(GreedySchedulerTest, ArgsortPadScan) {
   EXPECT_TRUE(executor_cache.getMostRecentKernelRuntime()->isSegmented());
 }
 
+// Based on SgLangMoETest.ComputeArgSort
+TEST_F(GreedySchedulerTest, Scatter) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  const int64_t size = 128;
+
+  auto tv0 = makeContigConcreteTensor({size}, DataType::Int);
+  fusion.addInput(tv0);
+
+  auto tv1 = zeros({IrBuilder::create<Val>(size)}, DataType::Int);
+  auto tv2 = arange(
+      fusion.zeroVal(DataType::Int),
+      IrBuilder::create<Val>(size, DataType::Int),
+      DataType::Int);
+  auto tv3 = scatter(tv1, 0, tv0, tv2);
```

Comment:
Do we still have the restriction that we need to have matching size on the scatter dimension across all three TVs?

Reply:
The scatter dimension doesn't need to be the same between the input and index.